packages/app/src/cli/commands/app/build.ts

```diff
@@ -52,12 +50,8 @@ export default class Build extends AppCommand {
       cmd_app_dependency_installation_skipped: flags['skip-dependencies-installation'],
     }))
 
-    const specifications = await loadLocalExtensionsSpecifications()
-    const app: AppInterface = await loadApp({
-      specifications,
-      directory: flags.path,
-      userProvidedConfigName: flags.config,
-    })
+    const {app} = await linkedAppContext({directory: flags.path, clientId: flags['client-id']})
```

Comment:
If you make it a mandatory part of the options for the thing you're calling, that'll help. Its not perfect but it forces the caller to make a choice at least. So:
```
// not this
{ optionalThing?: string } // resolves to string | undefined
// do this
{ optionalThing: string | undefined } // resolves to string | undefined
// because
callFn({}) can fail and tell you you didn't provide `optionalThing`
```